dask/array/linalg.py

```diff
@@ -62,7 +62,7 @@ def tsqr(data, name=None, compute_svd=False):
     if not (data.ndim == 2 and                    # Is a matrix
             len(data.chunks[1]) == 1):         # Only one column block
         raise ValueError(
-            "Input must have the following properties:\n"
+            "[tsqr] Input must have the following properties:\n"
```

Comment:
Users can get to this error by calling the `da.linalg.qr` function, but this term, `tsqr`, may not be well understood by them.

Reply:
Will keep a reference to tsqr, but also refer to qr and the relevant case.
```
    if not (data.ndim == 2 and                    # Is a matrix
            len(data.chunks[1]) == 1):         # Only one column block
        raise ValueError(
            "Input must have the following properties:\n"
            "  1. Have two dimensions\n"
            "  2. Have only one column of blocks\n\n"
            "Note: This function (tsqr) supports QR decomposition in the case of\n"
            "tall-and-skinny matrices (single column chunk/block; see qr)"
        )
```

Some duplication in `qr`:
```
    if len(a.chunks[1]) == 1:
        return tsqr(a, name)
    elif len(a.chunks[0]) == 1:
        return sfqr(a, name)
    else:
        raise NotImplemented(
            "qr currently supports only tall-and-skinny (single column chunk/block; see tsqr)\n"
            "and short-and-fat (single row chunk/block; see sfqr) matrices\n\n"
            "Consider use of the rechunk method. For example,\n\n"
            "x.rechunk({0: -1, 1: 'auto'}) or x.rechunk({0: 'auto', 1: -1})\n\n"
            "which rechunk one shorter axis to a single chunk, while allowing\n"
            "the other axis to automatically grow/shrink appropriately."
        )
```
... but ok because "pure functions".